pkg/controller/volume/attachdetach/metrics/metrics.go

```diff
@@ -134,3 +149,21 @@ func (collector *volumeInUseCollector) getVolumeInUseCount() nodeVolumeCount {
 	}
 	return nodeVolumeMap
 }
+
+// RecordADControllerVolumesMetric registers the amount of volumes, and their respective states, in the A/D Controller.
+func RecordADControllerVolumesMetric(pluginName, state string) {
```

Comment:
nit: These methods are called directly, which makes testing the caller difficult because these dependencies can't be easily be mocked out. Unless there is a good reason otherwise, this package should have an interface, and a concrete object that implements that interface.

This file is already like this, so not your fault, but you could clean it up while you're touching the code if you want.

Reply:
> nit: These methods are called directly, which makes testing the caller difficult because these dependencies can't be easily be mocked out.

From what I've seen around, this seems to be a common pattern with metrics (where tests are usually implemented as e2e). In any case, I refactored this file and left the changes in a separate commit to facilitate the review. Please let me know if I'm on the right track.